### PR TITLE
Channelz: track listen sockets in the server node

### PR DESCRIPTION
**Pull request by hcaseyal**
Similar to https://github.com/grpc/grpc/pull/19624, this PR changes channelz to track listen sockets in the server node instead of in the grpc_server.

### CODE REVIEW

File: src/core/lib/channel/channelz.cc [hunk 1]
```diff
@@ -382,17 +390,18 @@ grpc_json* ServerNode::RenderJson() {
   // ask CallCountingHelper to populate trace and call count data.
   call_counter_.PopulateCallCounts(json);
   json = top_level_json;
-  ChildRefsList listen_sockets;
-  grpc_server_populate_listen_sockets(server_, &listen_sockets);
-  if (!listen_sockets.empty()) {
+
```
Comment (markdroth) [inline src/core/lib/channel/channelz.cc]: Nit: Please remove blank lines within functions.

File: src/core/lib/channel/channelz.h [hunk 2]
```diff
@@ -58,12 +58,8 @@ namespace channelz {
 grpc_arg MakeParentUuidArg(intptr_t parent_uuid);
 intptr_t GetParentUuidFromArgs(const grpc_channel_args& args);
 
-// TODO(ncteisen), this only contains the uuids of the children for now,
```
Comment (markdroth) [inline src/core/lib/channel/channelz.h]: We should probably retain this TODO somewhere.  API-wise, we'll probably want to do this by adding a string parameter to all of the `AddChild*()` methods in both `ChannelNode` and `ServerNode`.

File: src/core/lib/channel/channelz.h [hunk 3]
```diff
@@ -58,12 +58,8 @@ namespace channelz {
...
-// since that is all that is strictly needed. In a future enhancement we will
-// add human readable names as in the channelz.proto
-typedef InlinedVector<intptr_t, 10> ChildRefsList;
-
 class SocketNode;
+class ListenSocketNode;
```
Comment (markdroth) [inline src/core/lib/channel/channelz.h]: This forward declaration doesn't seem necessary.

File: src/core/lib/channel/channelz.cc (hunk 1 above)
Comment (hcaseyal) [reply src/core/lib/channel/channelz.cc]: Done

**Comment by hcaseyal** [comment]
Test failures are existing / unrelated / infrastructural:
https://github.com/grpc/grpc/issues/19614
https://github.com/grpc/grpc/issues/19738

**Pull request merged**